src/libpmem2/map.c

```diff
@@ -138,28 +138,32 @@ static os_rwlock_t lock;
  * mapping_min - return min boundary for mapping
  */
 static size_t
-mapping_min(void *map)
+mapping_min(void *addr)
 {
-	return (size_t)pmem2_map_get_address(map);
+	struct pmem2_map *map = (struct pmem2_map *)addr;
+	return (size_t)map->addr;
 }
 
 /*
  * mapping_max - return max boundary for mapping
  */
 static size_t
-mapping_max(void *map)
+mapping_max(void *addr)
 {
-	return (size_t)pmem2_map_get_address(map) +
-		pmem2_map_get_size(map);
+	struct pmem2_map *map = (struct pmem2_map *)addr;
+	return (size_t)map->addr + map->content_length;
 }
 
+os_rwlock_t vm_rsv_lock;
```

Comment:
`static`
And move this alongside the other globals.

And ideally, you should put them in a single struct, like this:

```
static struct pmem2_state {
    static struct ravl_interval *range_map;
    static os_rwlock_t range_map_lock;
    os_rwlock_t reservation_lock;
} State;
```

Use longer descriptive names for globals. The first letter of a global should be capitalized.

Reply:
I've made vm_rsv_lock an extern,  range_map and range_map_lock variables are needed in map.c, the reservation_lock is needed in map_posix.c and map_windows.c files.

Reply:
Since reservation_lock was made internal to the vm_reservation API, i've moved only range_map and range_map_lock to pmem2_state struct.